hoomd_polymers/base/molecule.py

```diff
@@ -0,0 +1,328 @@
+from hoomd.md.force import Force as HForce
+from mbuild.lib.recipes import Polymer as mbPolymer
+
+from hoomd_polymers.base.base_types import FF_Types
+from hoomd_polymers.utils import check_return_iterable
+from .forcefield import find_xml_ff, apply_xml_ff, _validate_hoomd_ff
+
+
+class Molecule:
+    def __init__(self, n_mols, force_field: Union[Dict, List[HForce], str]=None, smiles=None, file=None, description=None,
+                 remove_hydrogens=False):
+        self.n_mols = check_return_iterable(n_mols)
+        self.force_field = force_field
+        self.smiles = smiles 
+        self.file = file 
+        self.description = description
+        self.remove_hydrogens = remove_hydrogens
+        self._mapping = None
+        self._mb_molecule = self._load()
+        self.gmso_molecule, self.topology_information = self._get_topology_information()
+        if self.force_field:
+            self._validate_force_field()
+        self._molecules = []
+        self._cg_molecules = []
+        self._generate()
+
+
+    @property
+    def molecules(self):
+        """List of all instances of the molecule"""
+        if self._cg_molecules:
+            return self._cg_molecules
+        return self._molecules
+    
+    @property
+    def mapping(self):
+        """Dictionary of particle index to bead mapping"""
+        return self._mapping
+
+    @mapping.setter
+    def mapping(self, mapping_array):
+        self._mapping = mapping_array
+
+    @property
+    def particle_types(self):
+        return self.topology_information["particle_types"]
+
+    @property
+    def pair_types(self):
+        return self.topology_information["pair_types"]
+
+    @property
+    def bond_types(self):
+        return self.topology_information["bond_types"]
+
+    @property
+    def angle_types(self):
+        return self.topology_information["angle_types"]
+
+    @property
+    def diherdal_types(self):
+        return self.topology_information["dihedral_types"]
+
+    @property
+    def improper_types(self):
+        return self.improper_types["improper_types"]
+    
+    def coarse_grain(self, beads=None, mapping=None):
+        for comp in self.molecules:
+            cg_comp = CG_Compound(comp, beads=beads, mapping=mapping)
+            self._cg_molecules.append(cg_comp)
+
+    def _load(self):
+        if self.file and isinstance(self.file, str): # Loading from file takes precedent over SMILES 
+            return mb.load(self.file)
+        elif self.smiles and isinstance(self.smiles, str):
+            return mb.load(self.smiles, smiles=True)
+        else:
+            raise ValueError(
+                    "Unable to load from ",
+                    f"File: {self.file}",
+                    f"SMILES: {self.smiles}"
+            )
+
+    def _generate(self):
+        for i in range(self.n_mols):
+            self._molecules.append(self._mb_molecule)
+
+    def _convert_to_gmso(self):
+        topology = from_mbuild(self._mb_molecule)
```

Comment:
We need to be careful here. Related to what we've talked about befre with some of the espaloma challenges is how new types of particles, bonds and angles might emerge once we polymerize a monomer.  For example, the PPS monomer doesn't have a C-S-C angle, but the polymer does.  Note, the monomer has a C-C-S angle, but that is different than a C-S-C angle.

So, it's possible that we aren't completely validating a FF if we just use the monomer, and the `topology_information' might return incomplete information.  Maybe we just pull the first compound from `self.molecules` and try using that?

Reply:
Yeah, that's a good point. I'll use the first compound from molecules list instead.